### PR TITLE
Update Go driver tests to make CI pass

### LINKED ISSUE #193: Update Go driver tests
The Go driver's test configuration is currently failing and neglected. We should update the configuration, skip flaky tests, and stop ignoring its failures.

DoD: CI should pass.

### PR DESCRIPTION
**Pull request by b1ron**
Closes #193.

### CODE REVIEW

**Comment by b1ron** [comment]
@AlekSi [unified](https://github.com/mongodb/mongo-go-driver/tree/master/mongo/integration/unified) tests seem to be the culprit here. I need more time to investigate further, I will create a separate issue for that. For now I will just remove `mongo/integration/unified`.

**Comment by AlekSi** [comment]
Sounds good

File: tests/mongo-go-driver.yml [hunk 1]
```diff
@@ -10,6 +10,7 @@ args:
   - ./x/mongo/driver/topology
 # - ./mongo/integration
   - ./mongo/integration/unified
+  - -count=1
```
Comment (AlekSi) [inline tests/mongo-go-driver.yml:13]: Please also add `-p=1`, `-parallel=1`, increase timeout above, and remove `-shuffle=on`. We will fix all that later

File: tests/mongo-go-driver.yml [hunk 2]
```diff
@@ -126,4 +141,4 @@ results:
       - go.mongodb.org/mongo-driver/examples/documentation_examples/TestDocumentationExamples/QueryToplevelFieldsExamples
       - go.mongodb.org/mongo-driver/examples/documentation_examples/TestDocumentationExamples/RunCommandExamples
       - go.mongodb.org/mongo-driver/examples/documentation_examples/TestDocumentationExamples/StableAPExamples
-      - go.mongodb.org/mongo-driver/examples/documentation_examples/TestDocumentationExamples/UpdateExamples
+      - go.mongodb.org/mongo-driver/examples/documentation_examples/TestDocumentationExamples/UpdateExamples
```
Comment (AlekSi) [inline tests/mongo-go-driver.yml:144]: Please configure your editor to add newlines at EOF

File: tests/mongo-go-driver.yml [hunk 3]
```diff
@@ -1,15 +1,15 @@
 ---
 runner: gotest
 args:
-  - -timeout=2m
-  - -shuffle=on
+  - -timeout=5m
+  - -parallel=1
```
Comment (AlekSi) [inline tests/mongo-go-driver.yml:5]: Let's add `-p=1` (number of packages tested in parallel; `-parallel` is for a number of tests in one package)

**Review by AlekSi — APPROVED** [review]

**Comment by noisersup** [comment]
It's nice to see mongo driver tests passing, thanks! :)

**Pull request merged**